src/main/java/com/teragrep/rlp_11/RelpProbeConfiguration.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * RELP Commit Latency Probe RLP-11
+ * Copyright (C) 2024 Suomen Kanuuna Oy
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ *
+ * Additional permission under GNU Affero General Public License version 3
+ * section 7
+ *
+ * If you modify this Program, or any covered work, by linking or combining it
+ * with other code, such other code is not for that reason alone subject to any
+ * of the requirements of the GNU Affero GPL version 3 as long as this Program
+ * is the same Program as licensed from Suomen Kanuuna Oy without any additional
+ * modifications.
+ *
+ * Supplemented terms under GNU Affero General Public License version 3
+ * section 7
+ *
+ * Origin of the software must be attributed to Suomen Kanuuna Oy. Any modified
+ * versions must be marked as "Modified version of" The Program.
+ *
+ * Names of the licensors and authors may not be used for publicity purposes.
+ *
+ * No rights are granted for use of trade names, trademarks, or service marks
+ * which are in The Program if any.
+ *
+ * Licensee must indemnify licensors and authors for any liability that these
+ * contractual assumptions impose on licensors and authors.
+ *
+ * To the extent this program is licensed as part of the Commercial versions of
+ * Teragrep, the applicable Commercial License may apply to this file if you as
+ * a licensee so wish it.
+ */
+package com.teragrep.rlp_11;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+
+public class RelpProbeConfiguration {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(RelpProbeConfiguration.class);
+    private static final int MAXIMUM_PORT = 65535;
+    private final Map<String, String> config;
+
+    public RelpProbeConfiguration(final Map<String, String> config) {
+        this.config = config;
+    }
+
+    public void validate() throws RelpProbeConfigurationError {
```

Comment:
object wide validators are not necessary, validate values within the configuration providing methods meaning lazy validation. this is because as the objects using the configuration are immutable and therefore must not get the values during the runtime but on the instantiation meaning the values must be present at the instatiation which must happen at the program start which will cause the program terminate anyway if values are not proper.